### DISCUSSION TITLE
logtail binary and libPluginBase.so huge size

### DISCUSSION THREAD

**Discussion by yu1019594136** [alibaba/ilogtail#582] [Help] [answered]
因为我的使用环境存储空间很有限，编译后发现logtail核心bianry和插件库都挺大的。
对于插件体积我暂时想到的是去掉一些我不需要的插件来缩减体积，但对于ilogtail这个核心binary我看社区版比企业版SLS的好像也大了不少，这个有什么办法吗？还是我需要对核心代码做裁剪？

社区版
$ls -lh output/
total 288M
**-rwxr-xr-x 1 root root 147M Dec 18 15:48 ilogtail**
-rw-r--r-- 1 root root    4 Dec 18 15:50 ilogtail_config.json
-rwxr-xr-x 1 root root  15K Dec 18 15:44 libPluginAdapter.so
**-rw-r--r-- 1 root root 129M Dec 18 15:49 libPluginBase.so**
-rwxr-xr-x 1 root root  13M Dec 18 15:49 logreceiver
drwxr-xr-x 2 root root 4.0K Dec 18 15:50 user_yaml_config.d

企业版
$ls -lh /usr/local/ilogtail/ilogtail_0.16.54
-rwxr-xr-x 1 root root 13M May 10  2021 /usr/local/ilogtail/ilogtail_0.16.54
-rwxr-xr-x 1 root  root   53M May 10  2021 libPluginBase.so

**Comment by yu1019594136** [comment, marked as answer]
看到了一个make dist打包方式，这个方式会对binary做strip操作。这样体积就减小一些了

![image](https://user-images.githubusercontent.com/7940391/208288521-89d1cd85-bf64-47ce-96f6-3c43dab2ffdf.png)

**Comment by messixukejia** [comment]
1、binary上述方法即可
2、插件暂时没有，如果需要，可以参与进来一起开发下。我们可以协助下。
